src/types/redis_json.cc

```diff
@@ -452,6 +452,11 @@ rocksdb::Status Json::numop(JsonValue::NumOpEnum op, const std::string &user_key
   if (!number_res || !number_res.GetValue().value.is_number()) {
     return rocksdb::Status::InvalidArgument("should be a number");
   }
+
+  if (!number_res.GetValue().value.is_int64() && !number_res.GetValue().value.is_double()) {
+    return rocksdb::Status::InvalidArgument("number out of range");
+  }
```

Comment:
I think it can be fixed in the JsonValue::NumOp. Here we can only fix for `value`, not the number inside the current json.